File: branches/1.1_before_major_refactor_2.0/java/core/src/org/daisy/urakawa/property/channel/ManagedAudioChannel.java
package org.daisy.urakawa.property.channel;

import org.daisy.urakawa.exception.MethodParameterIsNullException;
import org.daisy.urakawa.media.Media;
import org.daisy.urakawa.media.data.audio.ManagedAudioMedia;

/**
 *
 */
public class ManagedAudioChannel extends AudioChannel {
	/**
	 * @param chMgr
	 * @throws MethodParameterIsNullException
	 */
	public ManagedAudioChannel(ChannelsManager chMgr)
			throws MethodParameterIsNullException {
		super(chMgr);
	}

	@Override
	public boolean canAccept(Media m) throws MethodParameterIsNullException {
		if (!super.canAccept(m))
			return false;
		if (m instanceof ManagedAudioMedia)
			return true;
		return false;
	}
}

File: branches/sdk-core-structedit/property/channel/ManagedAudioChannel.cs
using System;
using System.Collections.Generic;
using System.Text;
using urakawa.media;
using urakawa.media.data.audio;

namespace urakawa.property.channel
{
    ///// <summary>
    ///// A <see cref="Channel"/> that only accepts <see cref="ManagedAudioMedia"/>
    ///// </summary>
    //public class ManagedAudioChannel : AudioChannel
    //{
    //    /// <summary>
    //    /// Determines if a given <see cref="Media"/> can be accepted by the channel,
    //    /// which it can if it is a <see cref="ManagedAudioMedia"/>
    //    /// </summary>
    //    /// <param name="m">The given media</param>
    //    /// <returns>A <see cref="bool"/> indicating if the given media can be accepted</returns>
    //    public override bool CanAccept(urakawa.media.Media m)
    //    {
    //        if (!base.CanAccept(m)) return false;
    //        if (!(m is ManagedAudioMedia)) return false;
    //        return true;
    //    }
    //}
}